src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,423 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import ring_theory.algebraic
+import topology.algebra.polynomial
+import analysis.calculus.mean_value
+import data.real.irrational
+import data.set.intervals.infinite
+import algebra.big_operators
+
+open_locale big_operators
+open polynomial real set finset
+
+/-!
+# Liouville's theorem
+
+This file contains the proof of Liouville's theorem stating that all Liouville numbers are
+transcendental.
+-/
+
+/--
+- a number is transcendental ↔ a number is not algebraic
+- a Liouville's number `x` is a number such that
+  for every natural number `n`, there is a `a/b ∈ ℚ - ℤ` such that `0 < |x - a/b| < 1/bⁿ`
+-/
+
+def liouville_number (x : ℝ) := ∀ n : ℕ, ∃ a b : ℤ,
```

Comment:
```suggestion
def is_liouville (x : ℝ) := ∀ n : ℕ, ∃ a b : ℤ,
```
I think it would make sense to put this in the namespace `real`, because it is a predicate on real numbers.

Reply:
@jjaassoonn Please add this definition in the `real` namespace.